Orleans.Providers.MongoDB/StorageProviders/Serializers/JsonGrainStateSerializer.cs

```diff
@@ -23,6 +23,8 @@ protected JsonGrainStateSerializer(JsonSerializer serializer)
             // values that are not equal to the system defaults.
             this.serializer.NullValueHandling = NullValueHandling.Include;
 
+            this.serializer.ObjectCreationHandling = ObjectCreationHandling.Replace;
```

Comment:
This line of code fixes my issue. Try running the code with and without and you will see the difference.
Can you pass a custom `JsonSerializer`? If yes the it makes this change backward compatible (I can clean the test code and merge to master).

Reply:
I recommend following the pattern just implemented in the main repo - adding a delegate for configuring the `JsonSerializerSettings`: https://github.com/dotnet/orleans/pull/5662

Reply:
PR has been updated.